server/server.rb

```diff
@@ -13,7 +19,82 @@ class Server < Roda
   else
     logger = Logger.new(STDOUT)
   end
+
   use Rack::CommonLogger, logger
+  use Rack::Attack
+  use Rack::Static, urls: { "/code" => "app/views/static/code.html" }
+  use TokenAuthentication, exclude: [
+      '/',
+      '/v1/ping',
+      '/v1/auth',
+      '/cb',
+      '/code',
+      '/oauth2/token',
+      '/v1/nodes',   #authorized using grid token
+      '/v1/nodes/*'
+    ],
+    soft_exclude: [
+      '/oauth2/authorize',
+      '/authenticate'
+    ],
+    allow_expired: [
+      '/authenticate'
+    ]
+
+  # Accessor to global config. Defaults are loaded from config/defaults.yml.
+  # Changing values in defaults.yml will not overwrite values in DB.
+  def self.config
+    return @config if @config
+    @config = Configuration
+    config_defaults.each do |key, value|
+      if @config[key].nil?
+        logger.debug "Setting configuration key '#{key}' using default '#{value}'"
+        @config[key] = value
+      end
+    end
+    @config
+  end
+
+  # Path to server application root
+  def self.root
+    @root ||= Pathname.new(File.dirname(__FILE__))
+  end
+
+  # Service root url
+  def self.root_url
+    @url ||= config[:root_url]
+  end
+
+  def self.name
+    @name ||= config[:server_name]
+  end
+
+  # Global logger
+  def self.logger
```

Comment:
Why not include `Logging` module? This seems a little bit duplicate implementation to me.